tests/integration/test_integration.py

```diff
@@ -742,6 +742,41 @@ def test_n_uniques_loss_between(engine, unique_table2, unique_table1, data):
     assert operation(req[0].test(engine).outcome)
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        (identity, ["col_int"], {0: (0.45, 0.55), 1: (0.45, 0.55)}, (0, 0)),
+        (negation, ["col_int"], {0: (0.35, 0.55), 1: (0.55, 0.6)}, (0, 0)),
+        (negation, ["col_int"], {0: (0.35, 0.55), 1: (0.35, 0.45)}, (0, 0)),
+        (
+            identity,
+            ["col_int", "col_varchar"],
+            {(0, "hi0"): (0.45, 0.55), (1, "hi0"): (0.25, 0.3), (1, "hi1"): (0.2, 0.3)},
+            (0, 0),
+        ),
+        (
+            negation,
+            ["col_int", "col_varchar"],
+            {
+                (0, "hi0"): (0.45, 0.55),
+                (1, "hi0"): (0.25, 0.3),
+            },
+            (0, 0),
+        ),
+        (identity, ["col_varchar"], {"hi0": (0.65, 0.85), "hi1": (0.1, 0.35)}, (0, 0)),
+        (negation, ["col_varchar"], {"hi0": (0.65, 0.85)}, (0, 0)),
+        (identity, ["col_varchar"], {"hi0": (0.65, 0.85)}, (0, 0.35)),
+        (negation, ["col_varchar"], {}, (0, 0.35)),
+    ],
+)
+def test_distribution_within(engine, distribution_table, data):
```

Comment:
We might want to reflect the new constraint name in the test function name, e.g. `test_categorical_bound_within`.